### ISSUE TITLE
Unable to find PETSc after new install.

### ISSUE THREAD

**Issue by tranner** [firedrakeproject/firedrake#1627]
Hi,

I think I've successfully installed firedrake on a fedora/anaconda system. 
[firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/4339471/firedrake-install.log).

I've downloaded the Helmholtz example (https://www.firedrakeproject.org/demos/helmholtz.py.html) as a test case but this doesn't run.

```
tr@comp-pc3135-fedora-vm:~/Software/firedrake ((HEAD)) (empty)
$ . ~/opt/source/firedrake/bin/activate                                                                               
tr@comp-pc3135-fedora-vm:~/Software/firedrake ((HEAD)) (empty)
$ which python
/home/tr/opt/source/firedrake/bin/python
tr@comp-pc3135-fedora-vm:~/Software/firedrake ((HEAD)) (empty)
$ python test.py 
Traceback (most recent call last):
  File "test.py", line 51, in <module>
    from firedrake import *
  File "/home/tr/opt/source/firedrake/src/firedrake/firedrake/__init__.py", line 19, in <module>
    import firedrake.petsc as petsc
  File "/home/tr/opt/source/firedrake/src/firedrake/firedrake/petsc.py", line 4, in <module>
    petsc4py.init(sys.argv)
  File "/home/tr/opt/source/firedrake/lib/python3.7/site-packages/petsc4py/__init__.py", line 42, in init
    PETSc = petsc4py.lib.ImportPETSc(arch)
  File "/home/tr/opt/source/firedrake/lib/python3.7/site-packages/petsc4py/lib/__init__.py", line 29, in ImportPETSc
    return Import('petsc4py', 'PETSc', path, arch)
  File "/home/tr/opt/source/firedrake/lib/python3.7/site-packages/petsc4py/lib/__init__.py", line 73, in Import
    module = import_module(pkg, name, path, arch)
  File "/home/tr/opt/source/firedrake/lib/python3.7/site-packages/petsc4py/lib/__init__.py", line 57, in import_module
    f, fn, info = imp.find_module(name, pathlist)
  File "/home/tr/miniconda3/lib/python3.7/imp.py", line 296, in find_module
    raise ImportError(_ERR_MSG.format(name), name=name)
ImportError: No module named 'PETSc'
```
Can anyone help?

**Comment by dham** [comment]
I'm a bit mystified. The log appears to show that module getting created in the petsc4py build process. Can you recreate the issue outside of the firedrake install process by running:
```python
import petsc4py
petsc4py.init()
```

**Comment by tranner** [comment]
I managed to find a bit more time to investigate. I was a bit suspicious about the call to `/home/tr/miniconda3/lib/python3.7/imp.py`. I've managed to find a way to install after removing all options to load conda. Once I'm gone through the installing stage I simply need to activate the appropriate venv (doesn't matter about a clashing conda env) and play with firedrake.

Thanks for the quite response and prompt for where to start looking.
